frontend/src/app/modules/common/back-routing/back-routing.service.ts

```diff
@@ -0,0 +1,63 @@
+// -- copyright
+// OpenProject is a project management system.
+// Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License version 3.
+//
+// OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+// Copyright (C) 2006-2013 Jean-Philippe Lang
+// Copyright (C) 2010-2013 the ChiliProject Team
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+//
+// See doc/COPYRIGHT.rdoc for more details.
+// ++
+
+import {Injectable, Injector} from '@angular/core';
+import {StateService} from "@uirouter/core";
+import {KeepTabService} from "core-components/wp-single-view-tabs/keep-tab/keep-tab.service";
+
+interface BackRouteOptions {
+  name:string;
+  params:{};
+  parent:string;
+}
+
+@Injectable()
+export class BackRoutingService {
+  public backRoute:BackRouteOptions;
+  private $state:StateService = this.injector.get(StateService);
+  private keepTab:KeepTabService = this.injector.get(KeepTabService);
+
+  constructor(protected injector:Injector) {
+  }
+
+  public goBack() {
+    if (this.backRoute) {
+      if (this.backRoute.parent === 'work-packages.list.details') {
```

Comment:
Do we want to route back to the current tab when using this button? I believe this might be confusing since switching between tabs in the full view will modify the behavior of the 'back' button. I do not have a strong opinion here.

If we decide to keep this, I would like this check moved into the keepTabService, something like `if (keepTab.relevantState(this.backroute.parent))` to avoid maintaining this differentiation here.

Reply:
I'm for keeping the tabs. I initially stumbled across the fact that they weren't taken over when I switched from full to details view.